payments/payment-methods/paypal/src/main/java/com/appcoins/payment_method/paypal/repository/PaypalRepository.kt

```diff
@@ -82,119 +82,72 @@ internal class PaypalRepositoryImpl @Inject constructor(
     ewt: String,
     walletAddress: String,
     token: String,
-  ): Boolean = paypalV2Api.cancelToken(
-    ewt = "Bearer $ewt",
-    walletAddress = walletAddress,
-    token = token,
-    paypalMetadataId = paypalHttpHeaderProvider.getMetadataId(),
-  ).isSuccessful
+  ): Boolean = restClient.post<Unit>(
+    path = "broker/8.20230522/gateways/paypal/billing-agreement/token/cancel",
+    header = mapOf(
+      "PayPal-Client-Metadata-Id" to paypalHttpHeaderProvider.getMetadataId(),
+      "authorization" to "Bearer $ewt"
+    ),
+    query = mapOf("wallet.address" to walletAddress),
+    body = token,
+  ).let { true }
 
   override suspend fun createBillingAgreement(
     ewt: String,
     walletAddress: String,
     token: String,
   ): BillingAgreement {
-    val billingAgreementResponse = paypalV2Api.createBillingAgreement(
-      paypalMetadataId = paypalHttpHeaderProvider.getMetadataId(),
-      ewt = "Bearer $ewt",
-      walletAddress = walletAddress,
-      token = token
-    )
-    return BillingAgreement(
-      uid = billingAgreementResponse.uid
+    val billingAgreementResponse = restClient.post<BillingAgreementResponse>(
+      path = "broker/8.20230522/gateways/paypal/billing-agreement/create",
+      header = mapOf(
+        "PayPal-Client-Metadata-Id" to paypalHttpHeaderProvider.getMetadataId(),
+        "authorization" to "Bearer $ewt"
+      ),
+      query = mapOf("wallet.address" to walletAddress),
+      body = token
     )
+    return BillingAgreement(uid = billingAgreementResponse.uid)
   }
 
   override suspend fun getCurrentBillingAgreement(
     ewt: String,
     walletAddress: String,
   ): BillingAgreement {
-    val response = paypalV2Api.getCurrentBillingAgreement(
-      paypalMetadataId = paypalHttpHeaderProvider.getMetadataId(),
-      ewt = "Bearer $ewt",
-      walletAddress = walletAddress,
-    )
-    return BillingAgreement(
-      uid = response.uid
+    val response = restClient.get<BillingAgreementResponse>(
+      path = "broker/8.20230522/gateways/paypal/billing-agreement",
+      header = mapOf(
+        "PayPal-Client-Metadata-Id" to paypalHttpHeaderProvider.getMetadataId(),
+        "authorization" to "Bearer $ewt"
+      ),
+      query = mapOf("wallet.address" to walletAddress),
     )
+    return BillingAgreement(uid = response.uid)
   }
 
   override suspend fun removeBillingAgreement(
     ewt: String,
     walletAddress: String,
-  ): Boolean =
-    paypalV2Api.removeBillingAgreement(
-      paypalMetadataId = paypalHttpHeaderProvider.getMetadataId(),
-      ewt = "Bearer $ewt",
-      walletAddress = walletAddress,
-    ).isSuccessful
+  ): Boolean = restClient.post<Unit>(
+    path = "broker/8.20230522/gateways/paypal/billing-agreement/cancel",
```

Comment:
This endpoint is returning 415

<-- 415 Unsupported Media Type https://api.dev.catappult.io/broker/8.20230522/gateways/paypal/billing-agreement/cancel?wallet.address=0x78A88BDdAaDA5E6c989d6A0826E7950f6177b5e6 (nullms)
null: HTTP/1.1 415 Unsupported Media Type

Reply:
Yes, this is known issue I pointed out in the PR description.
BE will fix it on their side.

Reply:
Ok. I understood in the daily that the BE had already fixed it.